letsencrypt/client/dns_authenticator.py

```diff
@@ -0,0 +1,261 @@
+"""DNS Authenticator"""
+import dns.query
+import dns.tsigkeyring
+import dns.update
+import dns.resolver
+
+import zope.interface
+
+from letsencrypt.acme import challenges
+
+from letsencrypt.client import achallenges
+from letsencrypt.client import constants
+from letsencrypt.client import errors
+from letsencrypt.client import interfaces
+
+def find_valid_key(tsig_keys, domain):
+    """Search provided TSIG key pairs.
+
+    Search the keypairs in `config.dns_tsig_keys` for one valid for the
+    provided `domain`.
```

Comment:
It would be a lot easier to read this function if you pass `config` as a first argument (documenting that `:type config: :class:letsencrypt.client.interfaces.IConfig`, with backquotes in right places), rather than `tsig_keys`. Comment would then also be redundant.